server/api/resolvers/requestGroupResolvers.ts

```diff
@@ -0,0 +1,139 @@
+import { Client, ClientInterface } from '../../database/models/clientModel'
+import { Request, RequestInterface } from '../../database/models/requestModel'
+import { RequestEmbeddingInterface, RequestType, RequestTypeInterface } from '../../database/models/requestTypeModel'
+import { RequestGroup, RequestGroupInterface } from '../../database/models/requestGroupModel'
+
+import { filterOpenRequestEmbeddings, nextRequestEmbeddingForRequestType } from './requestTypeResolvers'
+
+import { sessionHandler } from '../utils/session'
+
+const nextRequestEmbeddingForRequestGroup = async (requestGroup, session) => {
+    const nextRequestsForAllRequestTypes: Array<RequestEmbeddingInterface> = await Promise.all(requestGroup.requestTypes.map(async (requestTypeEmbedding) => {
+        const requestType = await RequestType.findById(requestTypeEmbedding._id).session(session)
+        return nextRequestEmbeddingForRequestType(requestType, session)
+    }))
+
+    if (nextRequestsForAllRequestTypes === null) {
+        return null
+    }
+
+    const sortedNextRequests = nextRequestsForAllRequestTypes
+        .sort((requestEmbedding1, requestEmbedding2) => {
+            if (requestEmbedding1 === null) {
+                return 1
+            }
+
+            if (requestEmbedding2 === null) {
+                return -1
+            }
+            
+            return requestEmbedding1.createdAt.getMilliseconds() - requestEmbedding2.createdAt.getMilliseconds()
+        })
+    
+    if (sortedNextRequests.length === 0) {
+        return null
+    }
+
+    return sortedNextRequests[0]
+}
+
+const requestGroupQueryResolvers = {
+    requestGroup: async (_, { _id }, ___): Promise<RequestGroupInterface> => {
+        return RequestGroup.findById(_id).exec()
```

Comment:
would you have to await here to get response back from RequestGroup.findById

Reply:
No. GraphQL resolvers handle awaiting Promises before returning the API response